moto/mediaconnect/models.py

```diff
@@ -0,0 +1,191 @@
+from __future__ import unicode_literals
+
+from collections import OrderedDict
+from uuid import uuid4
+
+from boto3 import Session
+from moto.core import BaseBackend, BaseModel
+from moto.mediaconnect.exceptions import NotFoundException
+
+
+class Flow(BaseModel):
+    def __init__(self, *args, **kwargs):
+        self.availability_zone = kwargs.get("availability_zone")
+        self.entitlements = kwargs.get("entitlements", [])
+        self.name = kwargs.get("name")
+        self.outputs = kwargs.get("outputs", [])
+        self.source = kwargs.get("source", {})
+        self.source_failover_config = kwargs.get("source_failover_config", {})
+        self.sources = kwargs.get("sources", [])
+        self.vpc_interfaces = kwargs.get("vpc_interfaces", [])
+        self.status = "STANDBY"  # one of 'STANDBY'|'ACTIVE'|'UPDATING'|'DELETING'|'STARTING'|'STOPPING'|'ERROR'
+        self._previous_status = None
+        self.description = None
+        self.flow_arn = None
+        self.egress_ip = None
+
+    def to_dict(self, include=None):
+        data = {
+            "availabilityZone": self.availability_zone,
+            "description": self.description,
+            "egressIp": self.egress_ip,
+            "entitlements": self.entitlements,
+            "flowArn": self.flow_arn,
+            "name": self.name,
+            "outputs": self.outputs,
+            "source": self.source,
+            "sourceFailoverConfig": self.source_failover_config,
+            "sources": self.sources,
+            "status": self.status,
+            "vpcInterfaces": self.vpc_interfaces,
+        }
+        if include:
+            new_data = {k: v for k, v in data.items() if k in include}
+            if "sourceType" in include:
+                new_data["sourceType"] = "OWNED"
+            return new_data
+        return data
+
+    def resolve_transient_states(self):
+        if self.status in ["STARTING"]:
+            self.status = "ACTIVE"
+        if self.status in ["STOPPING"]:
+            self.status = "STANDBY"
+        if self.status in ["UPDATING"]:
+            self.status = self._previous_status
```

Comment:
`self._previous_status` doesn't seem to be used anywhere else.
Wouldn't the status become "ACTIVE" after it's finished being updated? (I've never used this service, but that seems like the best value to use)

Reply:
I suppose it's not really relevant now, only after we implement `update_flow()`

Reply:
Correct, that's only necessary for the `update_` operations (which I am not going to implement in this PR, as I have no need for them currently). 

I did something similar for MediaLive, where I _did_ implement the `update_channel()` method. The reason I needed to keep track of `_previous_status` there is because it was important to return to whatever the channel state was before the update - and it could be either `IDLE` or `RUNNING`, because you can update a channel before or after it is started. 

Likewise, for MediaConnect, I'm expecting that you can perform update operations before or after starting a flow, so the `_previous_status` could potentially be one of `STANDBY` or `ACTIVE`.